### ISSUE TITLE
Add support for disabling max_runtime

### ISSUE THREAD

**Issue by hellais** [ooni/probe#893]
Currently in order to not have any limit to the max_runtime the user has to type some very large max_runtime value.

It should be possible for them to type 0 and for there not to be any limit to the max_runtime.

Maybe another iteration could include also a better UI for it.

**Comment by agrabeli** [comment]
Thank you @hellais for opening this ticket! I think many people would appreciate having an easy way to disable the max runtime (and test everything, or otherwise perform longer testing).

To make it more obvious (since I wonder whether people will imagine that they need to type 0 under "Test duration"), perhaps we can include this as an additional setting in the Websites card?

^^ @lorenzoPrimi

**Comment by hellais** [comment]
> perhaps we can include this as an additional setting in the Websites card?

I think we can probably just add a toggle switch next to the setting.

Maybe @holantonela can help us in designing this UI element.

**Comment by lorenzoPrimi** [comment]
So I will add an option called "Limit test duration" with a switch, if enabled it shows the "Test duration" inputbox.
Should this be on (like now with 90) or off by default? @hellais

**Comment by lorenzoPrimi** [comment]
Two more questions: 
does it makes sense that the user sets max_runtime to 90 and sees 120 in the test page?
What should be shown in case of no max runtime in test duration?

**Comment by hellais** [comment]
> does it makes sense that the user sets max_runtime to 90 and sees 120 in the test page?

Yes I agree this is not ideal. Do you have some thoughts on how we might make the runtime the user enters closer to the progress bar runtime?

> What should be shown in case of no max runtime in test duration?

How is the runtime currently determined? Is it possible for you to dynamically calculate the total runtime based on the number of inputs that you have retrieved?

In probe-cli (the desktop app), here is how it's calculate: https://github.com/ooni/probe-cli/blob/master/nettests/nettests.go#L193.
Basically you calculate the start time and every time you have measured a URL, you calculate the delta to determine the average time it took you to do one input.
You then multiply that by the number of remaining inputs.

**Comment by lorenzoPrimi** [comment]
> Yes I agree this is not ideal. Do you have some thoughts on how we might make the runtime the user enters closer to the progress bar runtime?
From what I understood these +30 seconds have to be there. Maybe we can set max_runtime to 60 when the user enters 90? @bassosimone 

> How is the runtime currently determined? Is it possible for you to dynamically calculate the total runtime based on the number of inputs that you have retrieved?
Sorry I asked the question the wrong way, I wanted to know what to show in the dashboard, before running the test.
Maybe we just remove the "X Sec" label?
Once the test is running I already have the algorithm to calculate duration based on the inputs

**Comment by hellais** [comment]
> From what I understood these +30 seconds have to be there. Maybe we can set max_runtime to 60 when the user enters 90?

I think that we should not alter this behaviour with such tricky code. I think we are better off explaining better with copy what is the meaning of this setting and clarifying that the ETAs are estimate times and not precise.
I think the meaning of the settings should not change, but maybe how we use it and compute the ETAs can be improved.

I would do this as a separate issue though, as it's out of scope for this I think.

> Sorry I asked the question the wrong way, I wanted to know what to show in the dashboard, before running the test.
> Maybe we just remove the "X Sec" label?
> Once the test is running I already have the algorithm to calculate duration based on the inputs

Can we compute the ETA in a deterministic way from the max runtime? Like max_runtime + 30 seconds?

If not, then let us just remove the ETA indication when the user changes the default max_runtime or sets it to infinite.

**Comment by lorenzoPrimi** [comment]
> Can we compute the ETA in a deterministic way from the max runtime? Like max_runtime + 30 seconds?

Yes max_runtime is not used at all, ETA is all about how many URLs will be sent to this probe. 1h is unrealistic as I believe 100 URLs are sent now.
